stdlib/public/core/LifetimeManager.swift

```diff
@@ -120,25 +119,27 @@ public func _withUnprotectedUnsafeMutablePointer<T, Result>(
 ///     type. If you need to mutate the argument through the pointer, use
 ///     `withUnsafeMutablePointer(to:_:)` instead.
 /// - Returns: The return value, if any, of the `body` closure.
+@_alwaysEmitIntoClient
 @inlinable
-public func withUnsafePointer<T, Result>(
+public func withUnsafePointer<T, E, Result>(
   to value: T,
-  _ body: (UnsafePointer<T>) throws -> Result
-) rethrows -> Result
+  _ body: (UnsafePointer<T>) throws(E) -> Result
+) throws(E) -> Result
 {
   return try body(UnsafePointer<T>(Builtin.addressOfBorrow(value)))
 }
-#else
-// TODO: This should be unified with non-embedded Swift.
-@inlinable
-public func withUnsafePointer<T, E, Result>(
+
+/// ABI: Historical withUnsafePointer(to:_:) rethrows, expressed as "throws",
```

Comment:
Should this be in a `#if !$Embedded` block, since there is no ABI to maintain?

Reply:
I wonder if we shouldn't put all these legacy untyped-throws implementations in a file that we can simply not built for non-ABI or new platforms.

Reply:
I'll leave this is it is in *this* PR, if you're okay with that, but yes, we want to decide how to organize the legacy implementations. I don't feel strongly about this, but let me just note that there is no impact for Embedded Swift even if we keep the legacy impl in -- unused functions do not contribute to binary outputs.